x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings/latest_findings_table.tsx

```diff
@@ -64,14 +61,13 @@ const FindingsTableComponent = ({
   return (
     <>
       <EuiBasicTable
-        data-test-subj={TEST_SUBJECTS.FINDINGS_TABLE}
         loading={loading}
-        error={error ? extractErrorMessage(error) : undefined}
-        items={data?.page || []}
+        data-test-subj={TEST_SUBJECTS.FINDINGS_TABLE}
+        items={items}
         columns={columns}
         pagination={pagination}
         sorting={sorting}
-        onChange={setTableOptions}
+        onChange={({ page, sort }) => !!sort && setTableOptions({ page, sort })}
```

Comment:
`sort` [won't](https://github.com/elastic/eui/blob/2547afede9d0f20f7c9d946cb64ef53e664f3abb/src/components/basic_table/basic_table.tsx#L434-L446) be `null`. it's really a type issue. 

we can do `2`, although a bit misleading since we need to model types to work with `undefined` when it never is. 

it works because `setTableOptions` is actually `setUrlQuery` which takes `Partial` and merges with previous. but now we have a divergence:
1. according to types, `sort` state is set by `setUrlQuery` because `sort` may be `undefined` 
2. in practice, `sort` is never `undefined` so it's not set by merging previous in `setUrlQuery` but directly from passed args. 

subtle difference, doesn't matter much, end up the same anway. but one is implicit (fall-through `undefined`) and the other is explicit (throw / check for `undefined`) 


so i guess it's 
1. types with `Required` and adding `throw` instead of `!!sort`
2. types with `CriteriaWithPagination` and just passing `setTableOptions` directly 

@ari-aviran - LMK if option `2` is still preferable, will do if needed. just figured i'd point out the rationale behind the approach i started with.

Reply:
What you did looks good 👍🏼